src/server.ts

```diff
@@ -38,6 +41,17 @@ export default class Server {
     this.app.use(GroupRouter);
   }
 
+  private configureAPM() {
+    const apmAgent = apm.start({
+      serviceName: 'group-service',
+      secretToken: config.apm.secretToken,
+      serverUrl: config.apm.serverUrl,
+      active: config.apm.isActive === 'true',
+      environment: config.service.environment,
+    });
+    if (apmAgent.isStarted()) console.log('APM Agent started');
```

Comment:
Is the apm.start is atomic operation? 
if not, it may go through this if statement and not step in

Reply:
this line should not have been committed. I deleted it.